pkg/scheduling/requirements.go

```diff
@@ -120,12 +120,65 @@ func (r Requirements) Compatible(requirements Requirements) (errs error) {
 		if operator := requirements.Get(key).Operator(); r.Has(key) || operator == v1.NodeSelectorOpNotIn || operator == v1.NodeSelectorOpDoesNotExist {
 			continue
 		}
-		errs = multierr.Append(errs, fmt.Errorf("key %s does not have known values", key))
+		errs = multierr.Append(errs, fmt.Errorf("label %q does not have known values%s", key, labelHint(r, key)))
 	}
 	// Well Known Labels must intersect, but if not defined, are allowed.
 	return multierr.Append(errs, r.Intersects(requirements))
 }
 
+// editDistance is an implementation of edit distance from Algorithms/DPV
+func editDistance(s, t string) int {
```

Comment:
Thoughts on moving to functional?

Reply:
Sure if we have a second use.  Any ideas for where we might also use it?

Reply:
Just going to resolve and merge for now, happy to move later if we get another use case.